warning: logical 'and' of mutually exclusive tests is always false [-Wlogical-op]

Seen with GCC 9:

```
editor/editor_themes.cpp: In function 'void editor_register_and_generate_icons(Ref<Theme>, bool, int, bool)':
editor/editor_themes.cpp:207:43: warning: logical 'and' of mutually exclusive tests is always false [-Wlogical-op]
  207 |  bool force_filter = !(p_thumb_size == 64 && p_thumb_size == 32); // we don't need filter with original resolution
      |                        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
```

@djrm It comes from your commit. Can you please fix it?

--- a/editor/editor_themes.cpp
+++ b/editor/editor_themes.cpp
@@ -204,7 +204,7 @@ void editor_register_and_generate_icons(Ref<Theme> p_theme, bool p_dark_theme = 
 		}
 
 	// generate thumb files with the given thumb size
-	bool force_filter = !(p_thumb_size == 64 && p_thumb_size == 32); // we don't need filter with original resolution
+	bool force_filter = p_thumb_size != 64 && p_thumb_size != 32; // we don't need filter with original resolution
 	if (p_thumb_size >= 64) {
 		float scale = (float)p_thumb_size / 64.0 * EDSCALE;
 		for (int i = 0; i < editor_bg_thumbs_count; i++) {

